extensions/webirc-3.3.md

```diff
@@ -0,0 +1,65 @@
+
+If forward and reverse DNS do not match, the IP addesss SHOULD be sent as the `hostname` and the unverified `hostname` SHOULD NOT be sent (see [Security Considerations](#security-considerations)). If the connection fails to the IRCD, the WebIRC gateway MUST return an `ERROR` with an appropriate message and terminate the connection.
+
+### Examples
+Generic format.
+
+    WEBIRC password user hostname ip
+
+IP address resolves to hostname.
+
+    WEBIRC hunter2 gatewayname kiwiirc.com 104.25.32.27
+
+IP address does not resolve to hostname.
+
+    WEBIRC hunter2 gatewayname 104.25.32.27 104.25.32.27
+
+Error from invalid password.
+
+    ERROR :Invalid WebIRC password
+
+## Use Cases
+Webchat applications that make the connection to the IRCD via their servers are the primary intended use case. WebIRC allows network and channel operators to utilize the full extent of moderation and security tools available by exposing the user IP addresses and hosts.
+
+Other use cases can include bouncer services that wish to pass user information to the IRC server.
+
+## Limitations
+WebIRC requires that the both WebIRC gateway and IRC server be configured to accept the connection in advance. Specific reccomendations are outlined below in the [Security Considerations](#security-considerations) section.
+
+## Security Considerations
+WebIRC allows anyone with a valid password to successfully connect to the IRC server and spoof any desired hostname. IRC servers SHOULD use secure passwords, SHOULD whitelist the IP addresses for the allowed WebIRC gateway servers, and SHOULD close any connections with an ERROR command that attempt to use the `WEBIRC` command not originating from a whitelisted IP address or using an incorrect password. Because the possibility for hostname spoofing exists, IRC servers MAY attempt to further validate or resolve hostnames or and match them to the IP addess. It is the responsibility of the IRC server to verify the authenticity of connecting users and perform additional security checks as they see fit. To assist network operators and prevent abuse, IRC servers SHOULD show when a WebIRC connection is in use and SHOULD provide the original host when possible.
+
+## Current Implementations
+Widespread adoption of WebIRC already exists for both clients and servers. Below is a non-exhaustive list of current implementions to demonstrate in-use instances.
+ - Clients
+  - [KiwiIRC](https://kiwiirc.com/docs/webirc)
+  - [Mibbit](https://wiki.mibbit.com/index.php/Webirc)
+  - [qwebirc](http://qwebirc.org/features)
+ - Servers
+  - [charybdis](https://github.com/charybdis-ircd/charybdis/blob/release/4/extensions/m_webirc.c)
```

Comment:
github urls pointing to a branch are unlikely to be permanent. Worth considering using a git hash?